### PR TITLE
Add the ability to run Spinnaker locally from a local git repo.

### PR DESCRIPTION
**Pull request by brandonnelson3**

### CODE REVIEW

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/DistributedDeployer.java [hunk 1]
```diff
@@ -400,6 +405,10 @@ public RemoteAction deploy(DistributedServiceProvider<T> serviceProvider,
   private <T extends Account> Set<Integer> reapOrcaServerGroups(AccountDeploymentDetails<T> details,
       SpinnakerRuntimeSettings runtimeSettings,
       DistributedService<Orca, T> orcaService) {
+    if (runtimeSettings.getServiceSettings(orcaService.getService()).getSkipLiveCycleManagement()) {
+      return Collections.EMPTY_SET;
```
Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/DistributedDeployer.java]: `return new Collections.emptySet()` is I think the preferred way

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitDeckService.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.List;
+
+@EqualsAndHashCode(callSuper = true)
+@Data
+@Component
+public class LocalGitDeckService extends DeckService implements LocalGitService<DeckService.Deck> {
+  String deskSettingsPath = "settings.js";
+  String deckPath = "~/.spinnaker/" + deskSettingsPath;
+
+  String startCommand = "export SETTINGS_PATH=" + deckPath + "\n"
```
Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitDeckService.java]:
```java
String.join("\n", "line1",
  "line2",
  "line3");
```

is what we've been doing in other places I think

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitDeckService.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      + "yarn start";
+
+  @Autowired
+  String gitRoot;
+
+  @Autowired
+  ArtifactService artifactService;
+
+  @Override
+  public ServiceSettings buildServiceSettings(DeploymentConfiguration deploymentConfiguration) {
```
Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitDeckService.java]: Does this match exactly what's in the super class? if so, we can skip this

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitEchoService.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  @Autowired
+  ArtifactService artifactService;
+
+  @Override
+  protected String getConfigOutputPath() {
+    return "~/.spinnaker";
+  }
+
+  @Override
+  public ServiceSettings buildServiceSettings(DeploymentConfiguration deploymentConfiguration) {
```
Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitEchoService.java]: Same question here as above

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitIgorService.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  ArtifactService artifactService;
+
+  @Override
+  protected String getConfigOutputPath() {
+    return "~/.spinnaker";
+  }
+
+  @Override
+  public List<Profile> getProfiles(DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
+    List<Profile> profiles = super.getProfiles(deploymentConfiguration, endpoints);
```
Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitIgorService.java]: I don't think this method needs to be overridden here

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitFront50Service.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  @Override
+  protected String getConfigOutputPath() {
+    return "~/.spinnaker";
+  }
+
+  @Override
+  public List<Profile> getProfiles(DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
+    List<Profile> profiles = super.getProfiles(deploymentConfiguration, endpoints);
+    generateAwsProfile(deploymentConfiguration, endpoints, getHomeDirectory()).ifPresent(p -> profiles.add(p));
+    return profiles;
```
Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitFront50Service.java]: Is this identical to the super?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/DistributedDeployer.java (hunk 1 above)
Comment (brandonnelson3) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/DistributedDeployer.java]: Looks to be return Collections.emptySet();

https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#emptySet--

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitDeckService.java (hunk 3 above)
Comment (brandonnelson3) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitDeckService.java]: buildServiceSettings is not implemented anywhere above this.

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitEchoService.java (hunk 4 above)
Comment (brandonnelson3) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitEchoService.java]: buildServiceSettings is not implemented anywhere above this.

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitFront50Service.java (hunk 6 above)
Comment (brandonnelson3) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitFront50Service.java]: Nope this is slightly different.

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitIgorService.java (hunk 5 above)
Comment (brandonnelson3) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/local/git/LocalGitIgorService.java]: Deleted.

**Pull request merged**